### ISSUE TITLE
git-find-large-files permission error

### ISSUE THREAD

**Issue by bkimminich** [larsxschneider/git-repo-analysis#16]
```
$ git-find-large-files
mktemp: failed to create directory via template ‘/tmpgit-find-large-files.XXXXXX’: Permission denied
```

I cloned https://github.com/larsxschneider/git-repo-analysis into a folder I have full access to and have it on my `PATH` as well. I tried this via Git Bash on Windows.

**Comment by MarkTwoFive** [comment]
The issue appears to be that ``$TMPDIR`` does not contain a trailing slash in git bash.  
Ensuring the existence of a trailing slash [here](https://github.com/larsxschneider/git-repo-analysis/blob/master/git-find-large-files#L43) might solve this issue. (Have not tried that yet.)

**Comment by Aalmann** [comment]
Got the same issue with Git bash on Windows. Definitely the missing trailing slash in line 43 (added it locally works). Probably a previous check for trailing slash should be added.

**Comment by bkimminich** [comment]
Thanks, that fixed the issue on that line in the script!

**Comment by larsxschneider** [comment]
Thanks for the report 🙇 😊 !

I think I fixed it here: https://github.com/larsxschneider/git-repo-analysis/commit/f41af8154fd375e50ec120d813b157b56e3e781d

Can you confirm that this fix works for you?

**Comment by bkimminich** [comment]
The problem is gone and now I get
```
/c/Data/GitHub/git-repo-analysis/git-find-large-files: line 62: look: command not found
/c/Data/GitHub/git-repo-analysis/git-find-large-files: line 61: look: command not found
```
but that is nothing your script is to blame for... 😁

**Comment by larsxschneider** [comment]
@bkimminich Oh. I assume you are running the script in Git for Windows? Then I need to find an alternative to the `look` command 👍

**Comment by MarkTwoFive** [comment]
Yeah, the ``look`` command is not installed on git-for-windows by default.  
If there's an alternative available on Windows which also works on Linux, I would favor using that.

**Comment by larsxschneider** [comment]
`look` performs a binary search on a sorted list of lines. Alternatively, we could use `grep` on platforms where `look` is not installed. That would be slower because `grep` would not perform a binary search but it would work.

**Comment by MarkTwoFive** [comment]
Is the performance difference typically significant? If so, a check could determine whether or not ``look`` is available and notify the user about the potential performance decrease by using grep when ``look`` can not be used.

**Comment by larsxschneider** [comment]
I fixed it here: 835119da15619d6a8457b449e429bf117c363a64

**Comment by MarkTwoFive** [comment]
👍 Confirmed working
